torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -7,13 +7,23 @@ namespace torch {
 namespace distributed {
 namespace rpc {
 
+std::mutex RRefContext::destroyedMutex_;
+bool RRefContext::destroyed_ = false;
+
 RRefContext& RRefContext::getInstance() {
-  static RRefContext context(RpcAgent::getDefaultRpcAgent());
-  return context;
+  // Leaky singleton to avoid module destructor races.
+  static RRefContext* context = new RRefContext(RpcAgent::getDefaultRpcAgent());
+  return *context;
 }
 
-void RRefContext::destroyInstance() {
-  RRefContext::getInstance().checkRRefLeaks();
+void RRefContext::destroyInstance(bool ignoreRRefLeak) {
+  {
+    std::lock_guard<std::mutex> lock(RRefContext::destroyedMutex_);
+    RRefContext::destroyed_ = true;
+  }
+  if (!ignoreRRefLeak) {
+    RRefContext::getInstance().checkRRefLeaks();
```

Comment:
Q: How long do we want to keep this bypass available? If we are to keep this "ignore leak" bypass for a while, it might be better to print out the leak but not throw exception.

Reply:
It will be there for a while. Yes, let me add the print.